### PR TITLE
add account progress feature

### PR DESCRIPTION
**Pull request by converge**
it can be used for external tools to follow account progress.

The main idea is provide data for https://github.com/converge/instapy-dashboard, but it can be used for other tools or even if you want to just SELECT the data directly to the DB.

I added it before start bots interaction, so we avoid the possibility of not collecting it when session ends.

---

Also updated our ERM:
captura de tela 2018-12-08 as 23 19 19

### CODE REVIEW

**Comment by sionking** [comment]
@converge this is more than welcome thanks.
Can you explain the dashboard feature a bit for non-web ppl like me ?

**Comment by SMarcinno** [comment]
We chatted on the issue I opened today... This is exactly what I programmed so far. I guess once this request is approved, I will update your solution with the Engagement-Rate functionality I am working on at the moment. Do you plan on adding diagrams of follower growth on your dashboard?

**Comment by converge** [comment]
hey @Smadman , just did it. Join us, we can work together on this PR.

I would like to test it more before approve it to make sure it's 100% correct.

hey @timgrossmann , @sionking , I made a quick video , it will be easier to understand what the PR is up to:

https://youtu.be/xuMGsXnxu-8

**Comment by HCWcoder** [comment]
How to make it work. On windows?

**Comment by converge** [comment]
hey @HCWcoder t, there is a command line instructions down bellow, near the green button.

**Comment by HCWcoder** [comment]
@converge 

```
C:\Users\JayPixx\Documents\GitHub\instapy-dashboard>npm install
[       ...........] - postinstall: sill install executeActi
> integer@2.1.0 install C:\Users\JayPixx\Documents\GitHub\instapy-dashboard\node_modules\integer
> node-gyp rebuild


C:\Users\JayPixx\Documents\GitHub\instapy-dashboard\node_modules\integer>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Users\JayPixx\AppData\Local\Programs\Python\Python37-32\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:289:12)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\JayPixx\Documents\GitHub\instapy-dashboard\node_modules\integer
gyp ERR! node -v v10.14.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! integer@2.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the integer@2.1.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\JayPixx\AppData\Roaming\npm-cache\_logs\2018-12-10T17_09_26_209Z-debug.log
```

Please help

**Comment by SMarcinno** [comment]
I just checked out your code.. why did you reprogram so many functionality? you can use functions like `followers_count, following_count = get_relationship_counts (browser, username, logger) ` 
and 
`web_address_navigator (browser, "https://www.instagram.com/"+ username +"/")` 
`number_of_posts = getUserData("graphql.user.edge_owner_to_timeline_media.count", browser)`

Errors are already handelt there and changes have to be made on one spot only.
If you want I can update it once it is merged to the master. We should avoid repeating code that does the same thing.

**Comment by converge** [comment]
I'll update it this weekend.

**Comment by SMarcinno** [comment]
> I'll update it this weekend.

what exactly do you want to update?

**Comment by converge** [comment]
I'll follow your suggestions @Smadman , you're correct about the ready to use methods.

File: instapy/util.py [hunk 1]
```diff
@@ -1641,4 +1641,33 @@ def has_any_letters(text):
...
+    Args:
+        :browser: web driver
+        :username: Account to be updated
+        :logger: library to log actions
+    """
+    logger.info('Saving account progress...')
+    followers, following = get_relationship_counts (browser, username, logger)
+    
+    # save profile total posts
+    posts = getUserData("graphql.user.edge_owner_to_timeline_media.count", browser)
```
Comment (timgrossmann) [inline instapy/util.py:1657]: I think we also have to check all the methods again, kind of weird that we have camelCased method names in here.

**Comment by timgrossmann** [comment]
@converge Could you please also add an entry to the CHANGELOG file before merging this? 😊

**Comment by timgrossmann** [comment]
@converge Excellent, thank you very much!

**Pull request merged**